File: src/cli/project/get.ts
import type { Command } from "commander";
import { getClient } from "../../lib/client.ts";
import { printError, printJson } from "../../lib/output.ts";
import { resolveProject } from "../../lib/resolvers.ts";

export function registerGet(project: Command): void {
  project
    .command("get")
    .description("Project summary: status, progress, lead, dates, milestones")
    .argument("<id>", "Project ID, slug, or name")
    .action(async (id: string) => {
      try {
        const client = getClient();
        const p = await resolveProject(client, id);
        const lead = await p.lead;
        const milestones = await p.projectMilestones();
        printJson({
          id: p.id,
          slugId: p.slugId,
          url: p.url,
          name: p.name,
          description: p.description,
          content: p.content,
          status: p.state,
          progress: p.progress,
          lead: lead ? { id: lead.id, name: lead.name } : null,
          startDate: p.startDate,
          targetDate: p.targetDate,
          milestones: milestones.nodes.map((m) => ({
            id: m.id,
            name: m.name,
            targetDate: m.targetDate,
          })),
        });
      } catch (err) {
        printError(err instanceof Error ? err.message : "Get failed");
      }
    });
}

File: internal/cli/project/get.go
package project

import (
	"context"

	"github.com/Khan/genqlient/graphql"
	"github.com/spf13/cobra"

	"github.com/shhac/lin/internal/cli/shared"
	apierrors "github.com/shhac/lin/internal/errors"
	"github.com/shhac/lin/internal/linear"
	"github.com/shhac/lin/internal/mappers"
	"github.com/shhac/lin/internal/output"
	"github.com/shhac/lin/internal/resolvers"
)

func registerGet(parent *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "get <id>...",
		Short: "Project summary: status, progress, lead, dates, milestones",
		Args:  cobra.MinimumNArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			getOne := func(client graphql.Client, id string) (any, error) {
				resolved, err := resolvers.ResolveProject(client, id)
				if err != nil {
					return nil, err
				}

				resp, err := linear.ProjectGet(context.Background(), client, resolved.ID)
				if err != nil {
					return nil, apierrors.ClassifyGraphQLError(err)
				}
				return mappers.MapProjectDetail(resp.Project), nil
			}
			return shared.RunGet(args, getOne, renderProjectCard)
		},
	}
	output.AllowPretty(cmd)

	parent.AddCommand(cmd)
}
